akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -116,7 +117,39 @@ private[akka] trait InboundContext {
  */
 private[akka] object AssociationState {
   def apply(): AssociationState =
-    new AssociationState(incarnation = 1, uniqueRemoteAddressPromise = Promise(), quarantined = Set.empty)
+    new AssociationState(incarnation = 1, uniqueRemoteAddressPromise = Promise(), quarantined = QuarantinedUidSet.empty)
+
+  object QuarantinedUidSet {
```

Comment:
Ok. Good idea to have such Map/Set. I have some nice ideas optimizing it even further if it becomes necessary.